src/main/scala/Core/CSR.scala

```diff
@@ -239,6 +239,7 @@ class CSR(implicit val conf: FlexpretConfiguration) extends Module {
       if (conf.externalInterrupt) {
         reg_msip(io.rw.thread) := data_in(3).asBool
       }
+      reg_in_interrupt(io.rw.thread) := data_in(2).asBool
```

Comment:
Hmm yes, I believe it does...

The reason we needed this was so a thread could clear it, but I guess the thread can also set it. From the `fp_int_on_expire`:

```
    if (ret) { \
        __ie_jmp_buf_active[hartid] = false; \
        clear_csr(CSR_STATUS, 0x04); \
        clear_csr(CSR_STATUS, 0x08); \
        set_csr(CSR_STATUS, 0x10); \
        goto cleanup; \
    } else { \
```

I guess it's possible to change the .scala code so it can only be cleared. Should I do that?

Reply:
I guess it is OK as it is